### PR TITLE
Develop API docs - configuration service

### PR DESCRIPTION
**Pull request by PavloEsco**

### CODE REVIEW

File: src/main/scripts/ci/after-success-generate-docs.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+SCALECUBE_CF_SERVICE_DOCS='/tmp/scalecube-repo/configuration-service/'
+
+git clone git@github.com:scalecube/scalecube.github.io.git /tmp/scalecube-repo
+git clone https://github.com/apidoc/apidoc.git /tmp/apidoc-repo
+
+docker build -t apidoc /tmp/apidoc-repo
+
+mkdir -p /tmp/docs-generated $SCALECUBE_CF_SERVICE_DOCS && rm -rf $SCALECUBE_CF_SERVICE_DOCS*
```
Comment (aharonha) [inline src/main/scripts/ci/after-success-generate-docs.sh]: What is CF?

Comment (skolomiiets) [reply src/main/scripts/ci/after-success-generate-docs.sh]: Configuration Service

Comment (aharonha) [reply src/main/scripts/ci/after-success-generate-docs.sh]:
```suggestion
 mkdir -p /tmp/docs-generated $SCALECUBE_CF_SERVICE_DOCS && rm -rf $SCALECUBE_CFG_SERVICE_DOCS*
```

File: src/main/scripts/ci/after-success-generate-docs.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bash
+
+SCALECUBE_CF_SERVICE_DOCS='/tmp/scalecube-repo/configuration-service/'
```
Comment (aharonha) [inline src/main/scripts/ci/after-success-generate-docs.sh]:
```suggestion
 SCALECUBE_CFG_SERVICE_DOCS='/tmp/scalecube-repo/configuration-service/'
```

File: src/main/scripts/ci/after-success-generate-docs.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+docker build -t apidoc /tmp/apidoc-repo
+
+mkdir -p /tmp/docs-generated $SCALECUBE_CF_SERVICE_DOCS && rm -rf $SCALECUBE_CF_SERVICE_DOCS*
+
+docker run -u $(id -u) \
+   -v $TRAVIS_BUILD_DIR/ApiDocs:/apidoc/docs \
+   -v /tmp/docs-generated:/apidoc/docs-generated \
+   -it apidoc -f ".*\\.apidoc$" -i "/apidoc/docs" -v -o "docs-generated"
+
+cp -R /tmp/docs-generated/* $SCALECUBE_CF_SERVICE_DOCS
```
Comment (aharonha) [inline src/main/scripts/ci/after-success-generate-docs.sh]:
```suggestion
 cp -R /tmp/docs-generated/* $SCALECUBE_CFG_SERVICE_DOCS
```

File: src/main/scripts/ci/after-success-generate-docs.sh [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+cd $SCALECUBE_CF_SERVICE_DOCS && \
```
Comment (aharonha) [inline src/main/scripts/ci/after-success-generate-docs.sh]:
```suggestion
 cd $SCALECUBE_CFG_SERVICE_DOCS && \
```

File: src/main/scripts/ci/after-success-generate-docs.sh (hunk 1 above)
Comment (aharonha) [inline src/main/scripts/ci/after-success-generate-docs.sh]:
```suggestion
 mkdir -p /tmp/docs-generated $SCALECUBE_CFG_SERVICE_DOCS && rm -rf $SCALECUBE_CF_SERVICE_DOCS*
```

**Review by aharonha — COMMENTED** [review]
CF -> CFG

File: src/main/scripts/ci/after-success-generate-docs.sh (hunk 2 above)
Comment (skolomiiets) [reply src/main/scripts/ci/after-success-generate-docs.sh]: Good

**Review by aharonha — DISMISSED** [review]

**Review by aharonha — APPROVED** [review]
:+1: 
@PavloPetrina I see you are still on it. I will review it again as needed

File: src/main/scripts/cd/release-generate-docs.sh [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+cd $TRAVIS_BUILD_DIR && git checkout master
+
+SCALECUBE_CFG_SERVICE='/tmp/scalecube-cfg-service'
+SCALECUBE_CFG_SERVICE_DOCS='/tmp/scalecube-repo/configuration-service/'
+RELEASE=$(mvn -q -Dexec.executable="echo" -Dexec.args='${project.version}' --non-recursive exec:exec)
+RELEASE_TEMPLATE='0.0.0-CURRENT'
+
+# clone needed repositories
+git clone git@github.com:scalecube/scalecube.github.io.git /tmp/scalecube-repo
+git clone git@github.com:scalecube/scalecube-configuration-service.git $SCALECUBE_CFG_SERVICE
```
Comment (aharonha) [inline src/main/scripts/cd/release-generate-docs.sh:14]: @skolomiiets is it needed? we already are cloning this (in travis)

File: src/main/scripts/cd/release-generate-docs.sh [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+# copy newly created documentation to http://scalecube.io/
+cp -R /tmp/docs-generated/* $SCALECUBE_CFG_SERVICE_DOCS
+
+# commit documentation
+for project in $SCALECUBE_CFG_SERVICE $SCALECUBE_CFG_SERVICE_DOCS; do
+    cd $project && \
+    git add . && \
+    git commit -m "Feature: updated configuration-service documentation" && \
+    git push
```
Comment (aharonha) [inline src/main/scripts/cd/release-generate-docs.sh:42]: @skolomiiets can push after all are added and committed?

File: src/main/scripts/cd/release-generate-docs.sh (hunk 5 above)
Comment (skolomiiets) [reply src/main/scripts/cd/release-generate-docs.sh:14]: @aharonha  We need 2 branches at the same time: release branch for generating documentation and develop branch for adding this documention.

File: src/main/scripts/cd/release-generate-docs.sh (hunk 6 above)
Comment (skolomiiets) [reply src/main/scripts/cd/release-generate-docs.sh:42]: @aharonha Generated documentation added to 2 repositories, after that changes are pushed.

**Review by dmytro-lazebnyi — APPROVED** [review]

**Pull request merged**